### PR TITLE
added functions for environments

### PR DESCRIPTION
**Pull request by m-kovalsky**

### CODE REVIEW

File: src/sempy_labs/_notebooks.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+):
+    """
+    Creates a new notebook within a workspace based on a Jupyter notebook hosted in the web.
+
+    Parameters
+    ----------
+    notebook_name : str
+        The name of the notebook to be created.
+    url : str
+        The url of the Jupyter Notebook (.ipynb)
```
Comment (eisber) [inline src/sempy_labs/_notebooks.py:82]: document special github fallback behavior

File: src/sempy_labs/_workspaces.py [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+        The `principal type <https://learn.microsoft.com/rest/api/power-bi/groups/add-group-user#principaltype>`_.
+    workspace : str, default=None
+        The name of the workspace.
+        Defaults to None which resolves to the workspace of the attached lakehouse
+        or if no lakehouse attached, resolves to the workspace of the notebook.
+    """
+
+    (workspace, workspace_id) = resolve_workspace_name_and_id(workspace)
+
+    role_names = ["Admin", "Member", "Viewer", "Contributor"]
```
Comment (eisber) [inline src/sempy_labs/_workspaces.py:163]: extract into utility function to avoid duplication

File: src/sempy_labs/_workspaces.py [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+        The name of the capacity.
+    workspace : str, default=None
+        The name of the Fabric workspace.
+        Defaults to None which resolves to the workspace of the attached lakehouse
+        or if no lakehouse attached, resolves to the workspace of the notebook.
+    """
+
+    (workspace, workspace_id) = resolve_workspace_name_and_id(workspace)
+
+    dfC = fabric.list_capacities()
```
Comment (eisber) [inline src/sempy_labs/_workspaces.py:213]: introduce resolve_capacity_id

**Review by eisber — CHANGES_REQUESTED** [review]

**Comment by LauJohansson** [comment]
I looks like there is more features than the title mentions - I have given a review on the environments part.
Thanks for adding my request.

**Review by eisber — APPROVED** [review]

**Pull request merged**